docs/how-to/contribute-code.rst

```diff
@@ -17,6 +17,9 @@ The recommended way of setting up your development environment for Briefcase is
 to use a `virtual environment <https://docs.python.org/3/library/venv.html>`__,
 and then install the development version of Briefcase and its dependencies:
 
+Clone Briefcase and create virtual environment
```

Comment:
Along with adding documentation for conditional coverage, I also wanted to make these docs easier to skim. So, I added a lot more headers so you can easily scroll down the page and understand what's included...on top of the TOC on the right.

Additionally, I did play around with adding a tl;dr section....but something was telling me you probably wouldn't like it....

![image](https://github.com/beeware/briefcase/assets/387848/15dc430d-a223-4da1-b8a6-5f7b6f13d01e)

Nonetheless, the motivation for such a section stems from the seemingly not insignificant number of contributors that seemingly don't do most of what's on this page...